torch/csrc/jit/passes/constant_propagation.cpp

```diff
@@ -66,6 +60,7 @@ void propagateNode(Node* n) {
   for (size_t i = 0; i < outputs.size(); ++i) {
     try {
       auto new_output = graph->insertConstant(outputs[i]);
+      new_output->setType(n->outputs()[i]->type());
```

Comment:
What is happening here? This line is highly suspicious because we rely on insertConstant setting the type correctly for constants and previously it would get set to NoneType for non constants.

Reply:
This didn't show up as a bug yet, but if there were builtin op that returned t? and it was constant-prop'd to None then schema matching would fail.

Reply:
Can you add a test that forces this failure to occur then? I also agree that if the type is needed to create a none constant, then we should be using a `createNone(TypePtr t)` method that forces it to be specified. Otherwise lots of other callsites will end up forgetting it and re-introducing the problem.

Reply:
Yes will do

Reply:
There is already a createNone() method - what is the change you would like exactly?